tools/run.py

```diff
@@ -0,0 +1,148 @@
+def run(command, catch_output=False, use_color=True):
+    sys.stdout.write(colorize('[RUN] {0}\n'.format(' '.join(command)), 'emph', use_color))
+    sys.stdout.flush()
+    if catch_output:
+        p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        return p.returncode, stdout, stderr
+    else:
+        return subprocess.call(command), None, None
+
+
+def get_common_parent(*directories):
+    parent = directories[0]
+    for dir in directories[1:]:
+        while os.path.relpath(dir, parent).startswith('..'):
+            old_parent, parent = parent, os.path.dirname(parent)
+            if old_parent == parent:
+                break
+    return parent
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Extra sanity test runner.')
+    parser.add_argument('--color',
+                        action='store_true',
+                        help='use ANSI colors')
+    parser.add_argument('targets',
+                        metavar='TARGET',
+                        nargs='*',
+                        help='targets')
+
+    args = parser.parse_args()
+
+    use_color = sys.stdout.isatty()
+    if args.color:
+        use_color = True
+
+    cwd = os.getcwd()
+    root = cwd
+    my_dir = '.'
+    try:
+        my_dir = os.path.dirname(__file__)
+        root = get_common_parent(root, my_dir)
+    except Exception as e:
+        pass
+
+    targets = list(args.targets)
+
+    container_name = 'ansible-test-{0}'.format(random.getrandbits(64))
+    output_filename = 'output-{0}.json'.format(random.getrandbits(32))
+
+    result = None
+    run([
+        'docker', 'run', '--detach',
+        '--workdir', os.path.abspath(cwd),
+        '--name', container_name,
+        CONTAINER,
+        '/bin/sh', '-c', 'sleep 50m',
+    ], use_color=use_color)
+    try:
+        run(['docker', 'cp', root, '{0}:{1}'.format(container_name, os.path.dirname(root))], use_color=use_color)
+        # run(['docker', 'exec', container_name, '/bin/sh', '-c', 'ls -lah ; pwd'])
+        command = ['docker', 'exec', container_name]
+        command.extend(['python3.7', os.path.relpath(os.path.join(my_dir, 'runner.py'), cwd)])
+        command.extend(['--cleanup', '--install-requirements', '--output', output_filename])
+        if use_color:
+            command.extend(['--color'])
+        command.extend(targets)
+        run(command, use_color=use_color)
+        dummy, result, stderr = run(['docker', 'exec', container_name, 'cat', output_filename], catch_output=True, use_color=use_color)
+        if stderr:
+            print(colorize('WARNING: {0}'.format(stderr.decode('utf-8').strip()), 'emph', use_color))
+    except Exception as e:
+        print(colorize('FATAL ERROR during execution: {0}'.format(e), 'emph', use_color))
+    finally:
+        try:
+            run(['docker', 'rm', '-f', container_name], use_color=use_color)
+        except Exception as dummy:
+            pass
```

Comment:
```suggestion
        except Exception as dummy:
            pass
```
maybe to print something in this case in this case would be better? It can be important to know that containers can't be removed.

Reply:
Will add something.